synapse/handlers/room_summary.py

```diff
@@ -812,9 +817,51 @@ async def get_room_summary(
 
                 room_summary["membership"] = membership or "leave"
         else:
-            # TODO federation API, descoped from initial unstable implementation
-            #      as MSC needs more maturing on that side.
-            raise SynapseError(400, "Federation is not currently supported.")
+            # Reuse the hierarchy query over federation
+            if remote_room_hosts is None:
+                raise SynapseError(400, "Missing via to query remote room")
+
+            (
+                room_entry,
+                children_room_entries,
+                inaccessible_children,
+            ) = await self._summarize_remote_room_hierarchy(
+                _RoomQueueEntry(room_id, remote_room_hosts),
+                suggested_only=True,
+            )
+
+            if room_entry:
+                room = room_entry.room
+                fed_room_id = room_entry.room_id
+
+                if fed_room_id == room_id:
```

Comment:
When would this not be true? (Or is just insurance?)

Reply:
I guess it can't happen here anymore. This is a leftover from me iterating over the list of rooms, which could potentially have child rooms. I guess in practice this can never be false, but it also doesn't hurt anything and makes bugs more obvious imo. I can remove it however.

Reply:
If we wanted to assert it I think we would want to do it in the shared code and raise an error on that case, instead of silently disregarding the bad data.

Reply:
Added an assert to the federation client now :3